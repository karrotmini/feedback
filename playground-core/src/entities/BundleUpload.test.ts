import { describe, expect } from 'vitest';

import {
  AppID,
  BundleUpload,
  BundleUploadID,
  UserProfileID,
} from '../entities';

describe('BundleUpload', test => {
  test('should have valid state after BundleUpload', () => {
    const id = BundleUploadID('TEST');
    const appId = AppID('TEST');
    const uploaderId = UserProfileID('TEST');

    const upload = new BundleUpload(id);
    upload.$publishEvent({
      aggregateId: id,
      aggregateName: 'BundleUpload',
      eventName: 'BundleUploaded',
      eventDate: new Date('2022-06-10').getTime(),
      eventPayload: {
        appId,
        uploaderId,
      },
    });

    expect(upload.$valid).toBe(true);
    expect(upload.$snapshot).toMatchSnapshot(`$snapshot version ${upload.snapshotVersion}`);
  });
});
