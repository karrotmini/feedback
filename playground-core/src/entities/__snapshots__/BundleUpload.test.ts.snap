// Vitest Snapshot v1

exports[`BundleUpload > should have valid state after BundleUpload > $snapshot version 1 1`] = `
{
  "appId": "TEST",
  "createdAt": 1654819200000,
  "tag": "",
  "uploaderId": "TEST",
}
`;
